clang/lib/CodeGen/CodeGenTypes.h

```diff
@@ -128,6 +128,15 @@ class CodeGenTypes {
   /// memory representation is usually i8 or i32, depending on the target.
   llvm::Type *ConvertTypeForMem(QualType T, bool ForBitField = false);
 
+  /// Check that size and abi alignment of given LLVM type matches size and
+  /// alignment of given AST type.
+  bool LLVMTypeLayoutMatchesAST(QualType ASTTy, llvm::Type *LLVMTy);
```

Comment:
I'm not sure we should be exposing LLVMTypeLayoutMatchesAST as a public API?  If any code cares, it's probably not calling ToMemory/FromMemory like it should.

Reply:
`ConstantEmitter::emitForMemory` is calling `LLVMTypeLayoutMatchesAST ` to determine if a big _BitInt constant needs to be split into individual bytes.

Reply:
In that case, please rename this to something that more accurately corresponds to that usage.

Reply:
Renamed to `typeRequiresSplitIntoByteArray`. Let me know if this is better.